app/scripts/controllers/network/createJsonRpcClient.test.js

```diff
@@ -0,0 +1,194 @@
+        { network: 'goerli', providerType: 'custom' },
+        ({ makeRpcCall }) => {
+          return makeRpcCall({
+            method: 'eth_chainId',
+          });
+        },
+      );
+
+      expect(chainId).toStrictEqual('0x5');
+    });
+  });
+
+  describe('eth_getTransactionByHash', () => {
+    const method = 'eth_getTransactionByHash';
+
+    testsForRpcMethodsThatCheckForBlockHashInResponse(method);
+
+    it("refreshes the block tracker's current block if it is less than the block number that comes back in the response", async () => {
+      await withMockedCommunications(
+        { providerType: 'custom' },
+        async (comms) => {
+          const request = { method };
+
+          comms.mockNextBlockTrackerRequest({ blockNumber: '0x100' });
+          // This is our request.
+          comms.mockRpcCall({
+            request,
+            response: {
+              result: {
+                blockNumber: '0x200',
+              },
+            },
+          });
+          comms.mockNextBlockTrackerRequest({ blockNumber: '0x300' });
+
+          await withClient(
+            { providerType: 'custom' },
+            async ({ makeRpcCall, blockTracker }) => {
+              await makeRpcCall(request);
+              expect(blockTracker.getCurrentBlock()).toStrictEqual('0x300');
+            },
+          );
+        },
+      );
+    });
+  });
+
+  describe('eth_getTransactionReceipt', () => {
+    const method = 'eth_getTransactionReceipt';
+
+    testsForRpcMethodsThatCheckForBlockHashInResponse(method);
+
+    it("refreshes the block tracker's current block if it is less than the block number that comes back in the response", async () => {
+      await withMockedCommunications(
+        { providerType: 'custom' },
+        async (comms) => {
+          const request = { method };
+
+          comms.mockNextBlockTrackerRequest({ blockNumber: '0x100' });
+          comms.mockRpcCall({
+            request,
+            response: {
+              result: {
+                blockNumber: '0x200',
+              },
+            },
+          });
+          comms.mockNextBlockTrackerRequest({ blockNumber: '0x300' });
+
+          await withClient(
+            { providerType: 'custom' },
+            async ({ makeRpcCall, blockTracker }) => {
+              await makeRpcCall(request);
+              expect(blockTracker.getCurrentBlock()).toStrictEqual('0x300');
+            },
+          );
+        },
+      );
+    });
+  });
+
+  describe('net_version', () => {
+    it('does not hit custom provider, instead returning the chain id that maps to the custom rpc network, as a decimal string', async () => {
+      const chainId = await withClient(
+        {
+          network: 'goerli',
+          providerType: 'custom',
+        },
+        ({ makeRpcCall }) => {
+          return makeRpcCall({
+            method: 'net_version',
+          });
+        },
+      );
+
+      expect(chainId).toStrictEqual('5');
```

Comment:
I would have thought this would fail since we hardcode chain_id: "0x1"

Reply:
Hmm, I'll take a look at this more closely.

Reply:
as mentioned above, fixed this now